skxray/speckle_analysis.py

```diff
@@ -0,0 +1,305 @@
+
+import logging
+logger = logging.getLogger(__name__)
+
+
+def roi_max_counts(images_sets, label_array):
+    """
+    Return the brightest pixel in any ROI in any image in the image set.
+
+    Parameters
+    ----------
+    images_sets : array
+        iterable of 4D arrays
+        shapes is: (len(images_sets), )
+
+    label_array : array
+        labeled array; 0 is background.
+        Each ROI is represented by a distinct label (i.e., integer).
+
+    Returns
+    -------
+    max_counts : int
+        maximum pixel counts
+    """
+    max_cts = 0
+    for img_set in images_sets:
+        for img in img_set:
+            max_cts = max(max_cts, maximum(img, label_array))
+    return max_cts
+
+
+def roi_pixel_values(image, labels, index=None):
+    """
+    This will provide intensities of the ROI's of the labeled array
+    according to the pixel list
+    eg: intensities of the rings of the labeled array
+
+    Parameters
+    ----------
+    image : array
+        image data dimensions are: (rr, cc)
+
+    labels : array
+        labeled array; 0 is background.
+        Each ROI is represented by a distinct label (i.e., integer).
+
+    index_list : list, optional
+        labels list
+        eg: 5 ROI's
+        index = [1, 2, 3, 4, 5]
+
+    Returns
+    -------
+    roi_pix : dict
+        intensities of the ROI's of the labeled array according
+        to the pixel list
+        {ROI 1 : intensities of the pixels of ROI 1, ROI 2 : intensities of
+         the pixels of ROI 2}
+    """
+
+    if labels.shape != image.shape:
+        raise ValueError("Shape of the image data should be equal to"
+                         " shape of the labeled array")
+    if index is None:
+        index = np.arange(1, np.max(labels) + 1)
+
+    roi_pix = {n: image[labels == n] for n in index}
+    return roi_pix, index
+
+
+def mean_intensity_sets(images_set, labels):
+    """
+    Mean intensities for ROIS' of the labeled array for different image sets
+
+    Parameters
+    ----------
+    images_set : array
+        images sets
+        shapes is: (len(images_sets), )
+        one images_set is iterable of 2D arrays dimensions are: (rr, cc)
+
+    labels : array
+        labeled array; 0 is background.
+        Each ROI is represented by a distinct label (i.e., integer).
+
+    Returns
+    -------
+    mean_intensity_list : list
+        average intensity of each ROI as a list
+        shape len(images_sets)
+
+    index_list : list
+        labels list for each image set
+
+    """
+    return tuple(map(list,
+                     zip(*[mean_intensity(im,
+                                          labels) for im in images_set])))
+
+
+def mean_intensity(images, labels, index=None):
+    """
+    Mean intensities for ROIS' of the labeled array for set of images
+
+    Parameters
+    ----------
+    images : array
+        Intensity array of the images
+        dimensions are: [num_img][num_rows][num_cols]
```

Comment:
Can you write this as would come back from shape `(num_img, num_rows, num_cols)`?